internal/auth0/tenant/flatten_test.go

```diff
@@ -0,0 +1,40 @@
+package tenant
+
+import (
+	"testing"
+
+	"github.com/auth0/go-auth0"
+	"github.com/auth0/go-auth0/management"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFlattenTenant(t *testing.T) {
+	mockResourceData := schema.TestResourceDataRaw(t, NewResource().Schema, map[string]interface{}{})
+
+	t.Run("it sets default values if remote tenant does not have them set", func(t *testing.T) {
+		tenant := management.Tenant{
+			IdleSessionLifetime: nil,
+			SessionLifetime:     nil,
+		}
+
+		err := flattenTenant(mockResourceData, &tenant)
+
+		assert.NoError(t, err)
+		assert.Equal(t, mockResourceData.Get("idle_session_lifetime"), idleSessionLifetimeDefault)
+		assert.Equal(t, mockResourceData.Get("session_lifetime"), sessionLifetimeDefault)
```

Comment:
It's not recommended to reference an implementation constant or var within the test as if these value change erroneously, the tests will still pass. 
```suggestion
		assert.Equal(t, mockResourceData.Get("idle_session_lifetime"), 72)
		assert.Equal(t, mockResourceData.Get("session_lifetime"), 168)
```

Reply:
I'm curious why you think so. These default values are subject to change and the tests assert that they equal the _defaults_, not some arbitrary number.

Reply:
Yes but they could change to a wrong default value and we wouldn't catch that in the tests.